internal/cloudapi/v2/handler.go

```diff
@@ -489,141 +489,143 @@ func imageTypeFromApiImageType(it ImageTypes, arch distro.Arch) string {
 }
 
 func (h *apiHandlers) GetComposeStatus(ctx echo.Context, id string) error {
-	jobId, err := uuid.Parse(id)
-	if err != nil {
-		return HTTPError(ErrorInvalidComposeId)
-	}
-
-	jobType, err := h.server.workers.JobType(jobId)
-	if err != nil {
-		return HTTPError(ErrorComposeNotFound)
-	}
-
-	if jobType == worker.JobTypeOSBuild {
-		var result worker.OSBuildJobResult
-		status, _, err := h.server.workers.OSBuildJobStatus(jobId, &result)
+	return h.server.EnsureJobChannel(func(ctx echo.Context, id string) error {
```

Comment:
I just wonder if a simple condition at the beginning for the handler methods wouldn't be cleaner. Using too many closures/lambdas feels a bit weird in Go. What do you think?

Reply:
I agree that the readability is a bit questionable.

I like the approach of using middleware method, because it easily allows us to stack multiple methods on top of each other.

I was considering moving the actual handler implementation into a private method, and then just calling it from the exported handler method as the last one. I think that this could improve the readability, while keeping the option to stack multiple methods on top of each other.

Reply:
I reworked the code a bit in the latest push, please have a look.

Reply:
Will have a look in a bit, sorry, a busy week.

Reply:
I like it, thanks! :)